nengo/spa/vocab.py

```diff
@@ -147,6 +149,17 @@ def add(self, key, p):
 
         The pointer value can be a SemanticPointer or a vector.
         """
+        if self.read_only:
+            if self.parent is None:
+                # If this vocabulary is not a subset vocabulary, throw an error
+                raise ValueError("Cannot add semantic pointer '%s' to "
+                                 "read-only vocabulary." % key)
+            else:
+                # If this vocabulary is a subset vocabulary, apply operation on
+                # parent vocab.
+                self.parent.add(key, p)
+                return
```

Comment:
Why is this fall-back here for the read_only case?  This allows this sort of thing to happen, which seems confusing to me:

```python
import nengo.spa

v = nengo.spa.Vocabulary(16)
v.extend(['A', 'B', 'C'])
v2 = v.create_subset(['A', 'B'])
v2.add('Z', v2.create_pointer())

print 'v.keys', v.keys
print 'v2.keys', v2.keys
```

```
v.keys ['A', 'B', 'C', 'Z']
v2.keys ['A', 'B']
```

It seems to me that it should raise a ValueError, rather than trying to create it in the parent.